src/components/DetailsOverlay/Toggle/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import { palette } from "@leafygreen-ui/palette";
+import Toggle from "@leafygreen-ui/toggle";
+import { Disclaimer } from "@leafygreen-ui/typography";
+import Cookie from "js-cookie";
+import { FORMAT_LOG_V2, PRETTY_PRINT_BOOKMARKS } from "constants/cookies";
+import { QueryParams } from "constants/queryParams";
+import { size } from "constants/tokens";
+import { useQueryParam } from "hooks/useQueryParam";
+import { FilterRow, StyledSubtitle } from "../styles";
+
+const { gray } = palette;
+
+interface BaseToggleProps {
+  ["data-cy"]?: string;
+  leftLabel?: string;
+  rightLabel?: string;
+  label: string;
+  value: boolean;
+  onChange: (checked: boolean) => void;
+}
+
+const BaseToggle: React.FC<BaseToggleProps> = ({
+  "data-cy": dataCy,
+  rightLabel = "ON",
+  leftLabel = "OFF",
+  label,
+  value,
+  onChange,
+}) => (
+  <FilterRow>
+    <StyledSubtitle> {label} </StyledSubtitle>
+    <ToggleWithLabel>
+      <StyledDisclaimer>{leftLabel}</StyledDisclaimer>
+      <Toggle
+        aria-labelledby="base-toggle"
+        checked={value}
+        data-cy={dataCy}
+        onChange={onChange}
+        size="small"
+      />
+      <StyledDisclaimer>{rightLabel}</StyledDisclaimer>
+    </ToggleWithLabel>
+  </FilterRow>
+);
+
+export const WrapToggle: React.FC = () => {
+  const [wrap, setWrap] = useQueryParam(QueryParams.Wrap, false);
+  return (
+    <BaseToggle
+      data-cy="wrap-toggle"
+      label="Wrap"
+      onChange={setWrap}
+      value={wrap}
+    />
+  );
+};
+
+export const CaseSensitiveToggle: React.FC = () => {
+  const [caseSensitive, setCaseSensitive] = useQueryParam(
+    QueryParams.CaseSensitive,
+    false
+  );
+  return (
+    <BaseToggle
+      data-cy="case-sensitive-toggle"
+      label="Case Sensitive"
+      onChange={setCaseSensitive}
+      value={caseSensitive}
+    />
+  );
+};
+
+export const LogV2Toggle: React.FC = () => {
+  const [logV2, setLogV2] = useState(Cookie.get(FORMAT_LOG_V2) === "true");
+  const onChange = (checked: boolean) => {
+    setLogV2(checked);
+    Cookie.set(FORMAT_LOG_V2, checked.toString(), { expires: 365 });
+  };
+
+  return (
+    <BaseToggle
+      data-cy="format-v2-toggle"
+      label="Format LogV2"
+      onChange={onChange}
+      value={logV2}
+    />
+  );
+};
```

Comment:
I spoke to @sanjogDB  lets omit this feature i think we will only support the resmoke v2 format since the legacy format isn't used anymore.